test/Npgsql.Tests/ReaderNewSchemaTests.cs

```diff
@@ -62,7 +62,7 @@ public void BaseColumnName()
                 {
                     var columns = reader.GetColumnSchema();
                     Assert.That(columns[0].BaseColumnName, Is.EqualTo("foo"));
-                    Assert.That(columns[1].BaseColumnName, Is.EqualTo("bar"));
+                    Assert.That(columns[1].BaseColumnName, Is.Null);
```

Comment:
Use `CommandBehavior.SchemaOnly | CommandBehavior.KeyInfo` and `Is.EqualTo("varchar")` should be `Is.Null`

Reply:
Should column info be returned without specifying `KeyInfo`?

/cc @roji

Reply:
With KeyInfo we shouldn't execute the database query to get the extra details - but we should still take whatever information we can from RowDescription. So we should definitely continue returning the ColumnName - as today - but it kinda makes sense for BaseColumnName to be null in that case.

Reply:
Did you mean without?

Reply:
Yep, corrected.

Reply:
@manandre are you saying that specifying SchemaOnly should prevent us from doing the query to get the extra information? If so, what's your reasoning?

The way I think about this, is that specifying KeyInfo is an opt-in for doing the extra information query, and SchemaOnly doesn't have any impact on that (it only means don't fetch actual result data, only schema).

Reply:
In my understanding, `BaseColumName` and `IsAliased` can only be correctly computed (i.e. `!= null`) from results of the extra information query (triggered in `KeyInfo` mode). Do I miss something?

Reply:
@manandre nope, that's my understanding as well. I just asked because you mentioned SchemaOnly above in your first comment - all good.